docs/dev-proc/source-code-structure.md

```diff
@@ -38,4 +38,48 @@ with following exceptions:
   happen it would be better to have it fixed by original author than the person
   that tries to upstream it some time later.
 
-Force-pushes to other branches are unconditionally forbidden.
+Force-pushes to  `<platform>/release` branches are unconditionally forbidden.
+
+## Merging guidelines
+
+We want to keep the history linear. The `rebase` merging strategy is desired.
+Merge commits in the code repositories are not allowed. The `rebase` strategy
+should be the only one available in the GitHub web UI.
+
+Is is, however, **strongly advised not to use GitHub web UI** to perform code
+merges. The `signed-off` tends to be dropped (even when using the `rebase`
+strategy), which is problematic for some projects (e.g. it makes the coreboot
+lint checks fail after merging from the UI).
+
+The procedure of merging is as follows:
+1. Review the code in GitHub.
+1. Make sure to receive at least one `Approve` in the review process.
+1. Make sure that **all** change requests are resolved.
+1. Merge the branch using git CLI. In case of merging the `feature` branch into
+   `develop` branch it may look as follows:
+
+    ```bash
+    $ git fetch dasharo
+    $ git checkout dasharo/<platform>/develop
+    $ git checkout -b <platform>/develop
```

Comment:
These two commands can be done as one.

Reply:
@krystian-hebel Please push improvement to this PR if you beleive it makes sense

Reply:
Done